src/box/index_def.c

```diff
@@ -211,6 +243,7 @@ index_stat_dup(const struct index_stat *src)
 void
 index_def_delete(struct index_def *index_def)
 {
+	index_def_unpin_coll_ids(index_def->key_def);
```

Comment:
This wouldn't work in a transaction:

```Lua
box.schema.space.create('test')
box.space.test:create_index('pk', {parts = {{2, 'string', collation = 'unicode_zh__zhuyin_s2'}}})
box.atomic(function()
    box.space.test:drop()
    box.space._collation.index.name:delete('unicode_zh__zhuyin_s2')
end)
```

We should probably move this code to `alter_space_do`.

Reply:
Fixed.